endoreg_db/models/media/video/create_from_file.py

```
@@ -67,16 +67,25 @@ def _create_from_file(
-        # 4. Prepare final storage path (for the raw file)
+            logger.warning("Video with hash %s already exists (UUID: %s). Checking file existence.", video_hash, existing_video.uuid)
+            
+            # Check if the existing video has a valid file
+            if existing_video.has_raw and existing_video.get_raw_file_path() and existing_video.get_raw_file_path().exists():
+                logger.warning("Video with hash %s already exists and file is present. Returning existing instance.", video_hash)
+                # Clean up transcoded file if it was created temporarily and is different from source
+                if transcoded_file_path != file_path and transcoded_file_path.exists():
+                    transcoded_file_path.unlink(missing_ok=True)
+                # Clean up the potentially empty output path if transcoding wasn't needed but path was provided
+                if transcoded_file_path == file_path and temp_transcoded_output_path.exists():
+                     temp_transcoded_output_path.unlink(missing_ok=True)
```

Flake8 (7.2.0)

[error] indentation is not a multiple of 4

(E111)